bin/sozo/src/commands/events.rs

```diff
@@ -59,59 +63,67 @@ impl EventsArgs {
 
             let (world_diff, provider, _) =
                 utils::get_world_diff_and_provider(self.starknet, self.world, &ws).await?;
-
             let provider = Arc::new(provider);
 
+            let latest_block = provider.block_number().await?;
             let from_block = if let Some(world_block) =
                 profile_config.env.as_ref().and_then(|e| e.world_block)
             {
-                Some(BlockId::Number(world_block))
+                world_block
             } else {
-                self.from_block.map(BlockId::Number)
-            };
-
-            let to_block = self.to_block.map(BlockId::Number);
-            let keys = self
-                .events
-                .map(|e| vec![e.iter().map(|event| starknet_keccak(event.as_bytes())).collect()]);
-
-            let event_filter = EventFilter {
-                from_block,
-                to_block,
-                address: Some(world_diff.world_info.address),
-                keys,
+                self.from_block.expect("from_block is required when not specified in environment")
```

Comment:
Avoid `expect` here, since this command returns a `Result`, you could use `anyhow` to bail an error that will more integrated in the output.